### ISSUE TITLE
[Feature]: Support launching multiple web servers

### ISSUE THREAD

**Issue by osiux** [microsoft/playwright#8206]
**Context:**
- GOOD Playwright Version: 1.13
- BAD Playwright Version: 1.14
- Operating System: Mac (didn't test other OS)
- Extra: N/A

**Code Snippet**

This used to work on 1.13 (updated to 1.14 option names). 

```javascript
const config: PlaywrightTestConfig = {
  webServer: [
    {
      command: 'yarn workspace @app/graphql start',
      port: 4000, 
      timeout: 60000,
      reuseExistingServer: !isCI,
    },
    {
      command: 'yarn workspace @app/frontend start',
      port: 3000, 
      timeout: 60000,
      reuseExistingServer: !isCI,
    }
  ]
};
```

**Describe the bug**

Previously we had the experimental `_launch` option to start a server for tests, and this options accepted an array of several servers to start. With 1.14 released it was renamed to `webServer` and only takes one server as a parameter. Maybe it's a weird use case but it was really helpful to be able to lift several servers. `baseUrl` fixture took the value of the last server launched.

Would it be possible to add this back?

**Comment by mauricedoepke** [comment]
This would be really helpful for us as well.

We would like to start
1) Our API server
2) Our Frontend "server" (next js)

Which are both needed to have a full environment to e2e test.

**Comment by pavliy** [comment]
Makes sense to me as well. In my case, I want to test authenticated scenarios with real auth services, but rest of tests - with auth fully mocked. So I thought starting separate servers for that - would totally make sense.

Right now the only alternative - to use separate playwright.config, which does not allow to have single test result.

**Comment by leobastiani** [comment]
https://github.com/microsoft/playwright/issues/11141

**Comment by vimtor** [comment]
Same thing here. One of the reasons I choose Playwright, is to test how different websites interact (e.g. create a company in an admin panel, see the company live in the web app)

**Comment by vimtor** [comment]
Maybe passing an array of `ports` could be a simpler alternative 😄

**Comment by weyert** [comment]
This looks interesting, I think it would be nice if the ports could  depend on another.

**Comment by jdkdev** [comment]
I need this as well!  In my case I was happy with passing in an array of web servers to start.

**Comment by rwoll** [comment]
Fixed by https://github.com/microsoft/playwright/pull/13779

**Comment by AndysonDK** [comment]
Any suggestions for any workarounds in the meantime?

**Comment by floehopper** [comment]
@AndysonDK I'm currently using something along the following lines:

```ts
// globalSetup.js
import { exec } from "child_process";

const globalSetup = async (config) => {
  const server = await exec(`${pathToExecutable}`);
  config.server = server;

  // add event handlers if necessary, e.g.
  // server.stdout.on("data", (data) => {})

  // add check for server started if necessary
}

// globalTeardown.js
const globalTeardown = async (config) => {
  if (config.server) await config.server.kill();
};
```

**Comment by rwoll** [comment]
Re-openeing and moving to v1.23; this was implemented atop plugins which will not be in 1.22.
